hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/StreamValidationOp.java

```diff
@@ -55,44 +62,96 @@ public class StreamValidationOp extends UtilOp {
     private static final long MIN_GZIP_SIZE_IN_BYTES = 26;
     private static final String ERROR_PREFIX = "\n  - ";
 
-    private static final List<RecordStreamValidator> STREAM_VALIDATORS = List.of(
+    private static final List<RecordStreamValidator> RECORD_STREAM_VALIDATORS = List.of(
             new BlockNoValidator(),
             new TransactionBodyValidator(),
             new ExpiryRecordsValidator(),
             new BalanceReconciliationValidator(),
             new TokenReconciliationValidator());
 
+    private static final List<BlockStreamValidator.Factory> BLOCK_STREAM_VALIDATOR_FACTORIES = List.of();
+
+    public static void main(String[] args) {}
+
     @Override
     protected boolean submitOp(@NonNull final HapiSpec spec) throws Throwable {
-        // Prepare streams for validators
+        // Prepare streams for record validators that depend on querying the network and hence
+        // cannot be run after submitting a freeze
         allRunFor(
                 spec,
                 // Ensure the CryptoTransfer below will be in a new block period
                 sleepFor(MAX_BLOCK_TIME_MS + BUFFER_MS),
                 cryptoTransfer((ignore, b) -> {}).payingWith(GENESIS),
                 // Wait for the final record file to be created
                 sleepFor(2 * BUFFER_MS));
-        // Validate the streams
-        readMaybeStreamDataFor(spec)
+        // Validate the record streams
+        final AtomicReference<RecordStreamAccess.Data> dataRef = new AtomicReference<>();
+        readMaybeRecordStreamDataFor(spec)
                 .ifPresentOrElse(
                         data -> {
-                            final var maybeErrors = STREAM_VALIDATORS.stream()
+                            final var maybeErrors = RECORD_STREAM_VALIDATORS.stream()
                                     .flatMap(v -> v.validationErrorsIn(data))
+                                    .peek(t -> log.error("Record stream validation error!", t))
                                     .map(Throwable::getMessage)
                                     .collect(joining(ERROR_PREFIX));
                             if (!maybeErrors.isBlank()) {
                                 throw new AssertionError(
                                         "Record stream validation failed:" + ERROR_PREFIX + maybeErrors);
                             }
+                            dataRef.set(data);
                         },
                         () -> Assertions.fail("No record stream data found"));
+        // Freeze the network
+        allRunFor(
+                spec,
+                freezeOnly().payingWith(GENESIS).startingIn(2).seconds(),
+                // Wait for the final stream files to be created
+                sleepFor(8 * BUFFER_MS));
```

Comment:
This didn't seem to work for me. I had to send one more transaction to get the final record file to close, so I added a crypto transfer of one tinybar here. Maybe I did something wrong...

Reply:
When platform goes to `FREEZE_COMPLETE` it should close records or blocks from the logic. Not sure why we need one more transfer 🤔

Reply:
When you say didn't seem to work , what do you mean @mhess-swl ? Did you try to enable in CI and that didn't work ?

Reply:
The test would fail when I run it on my machine, because of the unclosed record file

Reply:
Ohh .. I have never seen this. We could take a look. May be something is missing